src/libbson/src/jsonsl/jsonsl.c

```diff
@@ -1149,8 +1149,15 @@ void jsonsl_jpr_match_state_init(jsonsl_t jsn,
         return;
     }
     jsn->jprs = (jsonsl_jpr_t *)malloc(sizeof(jsonsl_jpr_t) * njprs);
+    if (!jsn->jprs) {
+        return;
+    }
     jsn->jpr_count = njprs;
     jsn->jpr_root = (size_t*)calloc(1, sizeof(size_t) * njprs * jsn->levels_max);
+    if (!jsn->jpr_root) {
+        free(jsn->jprs);
+        return;
+    }
```

Comment:
```suggestion
    jsn->jpr_root = (size_t *) bson_malloc0 (sizeof (size_t) * njprs * jsn->levels_max);
```

Similarly, use `bson_malloc0` in place of `calloc`. Suggest updating other `malloc` / `free` calls in this file.

Reply:
OK. That makes sense. Should I do that as a drive-by in this PR, or a separate PR?

Reply:
I think it is OK to include in this PR. Though the other `malloc` calls check their returns, I think it is a related enough change to include in this PR.